client/gossipEndpointDiscoverer.go

```diff
@@ -1,84 +1,133 @@
-	GossipSeeds               []*url.URL
-	NodePreference            NodePreference
-	SkipCertificateValidation bool
-	httpClient                *http.Client
-}
+	gossipApi "github.com/EventStore/EventStore-Client-Go/protos/gossip"
+	"github.com/EventStore/EventStore-Client-Go/protos/shared"
+	"google.golang.org/grpc"
+)
 
-func NewGossipEndpointDiscoverer() *GossipEndpointDiscoverer {
-	transport := &http.Transport{
-		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
-	}
-	client := &http.Client{Transport: transport}
-	return &GossipEndpointDiscoverer{
-		MaxDiscoverAttempts:       10,
-		NodePreference:            NodePreference_Leader,
-		SkipCertificateValidation: false,
-		httpClient:                client,
+func allowedNodeState() []gossipApi.MemberInfo_VNodeState {
+	return []gossipApi.MemberInfo_VNodeState{
+		gossipApi.MemberInfo_Follower,
+		gossipApi.MemberInfo_Leader,
+		gossipApi.MemberInfo_ReadOnlyLeaderless,
+		gossipApi.MemberInfo_PreReadOnlyReplica,
 	}
 }
 
-// Discover will discover nodes via performing a gossip over HTTP and then picking the best candidate to connect to
-func (discoverer *GossipEndpointDiscoverer) Discover() (*MemberInfo, error) {
-	if len(discoverer.GossipSeeds) == 0 {
-		return nil, errors.New("There are no gossip seeds")
-	}
-	discoverer.GossipSeeds = shuffleGossipSeeds(discoverer.GossipSeeds)
-	gossipIndex := 0
-	for attempt := 1; attempt <= discoverer.MaxDiscoverAttempts; attempt++ {
-		if gossipIndex >= len(discoverer.GossipSeeds) {
-			gossipIndex = 0
+func DiscoverNode(conf *Configuration) (*grpc.ClientConn, error) {
+	var connection *grpc.ClientConn = nil
+	attempt := 1
+
+	if conf.DnsDiscover || len(conf.GossipSeeds) > 0 {
+		var candidates []string
+
+		if conf.DnsDiscover {
+			candidates = append(candidates, conf.Address)
+		} else {
+			for _, seed := range conf.GossipSeeds {
+				candidates = append(candidates, seed.Host)
+			}
 		}
-		gossipSeed := discoverer.GossipSeeds[gossipIndex]
-		gossipIndex++
-		log.Printf("[info] attempting to gossip via %+v", gossipSeed)
-		member, err := discoverEndPoint(gossipSeed, discoverer.httpClient, discoverer.NodePreference)
-		if err != nil {
-			if attempt == discoverer.MaxDiscoverAttempts {
-				return nil, fmt.Errorf("Failed to discover any cluster node members via gossip. Maximum number of attempts reached. %s", err.Error())
+
+		shuffleCandidates(candidates)
+
+		for attempt <= conf.MaxDiscoverAttempts {
+			log.Printf("[info] discovery attempt %v/%v", attempt, conf.MaxDiscoverAttempts)
+			for _, candidate := range candidates {
+				log.Printf("[info] Attempting to gossip via %s", candidate)
+				connection, err := CreateGrpcConnection(conf, candidate)
+				if err != nil {
+					log.Printf("[warn] Error when creating a grpc connection for candidate %s", candidate)
+
+					continue
+				}
+
+				client := gossipApi.NewGossipClient(connection)
+				context, cancel := context.WithTimeout(context.Background(), time.Duration(conf.GossipTimeout))
```

Comment:
```suggestion
				context, cancel := context.WithTimeout(context.Background(), time.Duration(conf.GossipTimeout)*time.Second)
```
The `GossipTimeout` is defined in seconds and is set to `5` by default, which means that the cluster discovery currently times out in 5 milliseconds